datadog/resource_datadog_dashboard.go

```diff
@@ -4344,6 +4361,183 @@ func buildTerraformServiceLevelObjectiveDefinition(datadogDefinition datadogV1.S
+				},
+			},
+		},
+		"title": {
+			Description: "The title of the widget.",
+			Type:        schema.TypeString,
+			Optional:    true,
+		},
+		"title_size": {
+			Description: "The size of the widget's title. Default is 16.",
+			Type:        schema.TypeString,
+			Optional:    true,
+		},
+		"title_align": {
+			Description:  "The alignment of the widget's title. One of `left`, `center`, or `right`.",
+			Type:         schema.TypeString,
+			ValidateFunc: validators.ValidateEnumValue(datadogV1.NewWidgetTextAlignFromValue),
+			Optional:     true,
+		},
+		"time":      getDeprecatedTimeSchema(),
+		"live_span": getWidgetLiveSpanSchema(),
+		"custom_link": {
+			Description: "Nested block describing a custom link. Multiple `custom_link` blocks are allowed with the structure below.",
+			Type:        schema.TypeList,
+			Optional:    true,
+			Elem: &schema.Resource{
+				Schema: getWidgetCustomLinkSchema(),
+			},
+		},
+	}
+}
+
+func getGeomapRequestSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		// A request should implement exactly one of the following type of query
+		"q":         getMetricQuerySchema(),
+		"log_query": getApmLogNetworkRumSecurityQuerySchema(),
+		"rum_query": getApmLogNetworkRumSecurityQuerySchema(),
+	}
+}
+
+func buildDatadogGeomapDefinition(terraformDefinition map[string]interface{}) *datadogV1.GeomapWidgetDefinition {
+	datadogDefinition := datadogV1.NewGeomapWidgetDefinitionWithDefaults()
+	// Required params
+	terraformRequests := terraformDefinition["request"].([]interface{})
+	datadogDefinition.Requests = *buildDatadogGeomapRequests(&terraformRequests)
+
+	if style, ok := terraformDefinition["style"].([]interface{}); ok && len(style) > 0 {
+		if v, ok := style[0].(map[string]interface{}); ok && len(v) > 0 {
+			datadogDefinition.Style = buildDatadogGeomapRequestStyle(v)
+		}
+	}
+
+	if view, ok := terraformDefinition["view"].([]interface{}); ok && len(view) > 0 {
+		if v, ok := view[0].(map[string]interface{}); ok && len(v) > 0 {
+			datadogDefinition.View = buildDatadogGeomapRequestView(v)
+		}
+	}
+
+	// Optional params
+	if v, ok := terraformDefinition["custom_link"].([]interface{}); ok && len(v) > 0 {
+		datadogDefinition.SetCustomLinks(*buildDatadogWidgetCustomLinks(&v))
+	}
+
+	if v, ok := terraformDefinition["title"].(string); ok && len(v) != 0 {
+		datadogDefinition.SetTitle(v)
+	}
+
+	if v, ok := terraformDefinition["title_size"].(string); ok && len(v) != 0 {
+		datadogDefinition.SetTitleSize(v)
+	}
+
+	if v, ok := terraformDefinition["title_align"].(string); ok && len(v) != 0 {
+		datadogDefinition.SetTitleAlign(datadogV1.WidgetTextAlign(v))
+	}
+
+	if v, ok := terraformDefinition["time"].(map[string]interface{}); ok && len(v) > 0 {
+		datadogDefinition.Time = buildDatadogWidgetTime(v)
+	} else if ls, ok := terraformDefinition["live_span"].(string); ok && ls != "" {
+		datadogDefinition.Time = &datadogV1.WidgetTime{
+			LiveSpan: datadogV1.WidgetLiveSpan(ls).Ptr(),
+		}
+	}
```

Comment:
If we decide to remove the `time` part of the schema, we also need to remove this and just keep whats in the `else`

Reply:
Done in commit [82940d4](https://github.com/DataDog/terraform-provider-datadog/pull/954/commits/82940d4ce6e86936c65dd22e85b98ba68bcc21ef).